packages/pds/tests/db.test.ts

```diff
@@ -168,6 +168,42 @@ describe('db', () => {
     })
   })
 
+  describe('transaction advisory locks', () => {
+    if (db.dialect !== 'pg') {
+      return
+    }
+
+    it('allows locks in txs to run sequentially', async () => {
+      for (let i = 0; i < 100; i++) {
+        await db.transaction(async (dbTxn) => {
+          const locked = await dbTxn.txAdvisoryLock(1234)
+          expect(locked).toBe(true)
+        })
+      }
+    })
+
+    it('locks block between txns', async () => {
+      let resolve: () => void
+      const promise = new Promise<void>((res) => (resolve = res))
```

Comment:
We have a `createDeferrable()` util in common-web that might be more convenient for you here.